sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteBundlesToFiles.java

```diff
@@ -234,17 +245,28 @@ public void processElement(
 
     try {
       writer.write(element.getValue());
-    } catch (Exception e) {
-      // Discard write result and close the write.
+    } catch (BigQueryRowSerializationException e) {
       try {
-        writer.close();
-        // The writer does not need to be reset, as this DoFn cannot be reused.
-      } catch (Exception closeException) {
-        // Do not mask the exception that caused the write to fail.
-        e.addSuppressed(closeException);
+        badRecordRouter.route(
+            outputReceiver, element, coder, e, "Unable to Write BQ Record to File");
+      } catch (Exception e2) {
+        cleanupWriter(writer, e2);
       }
-      throw e;
+    } catch (IOException e) {
```

Comment:
This used to catch `Exception`. Is there a reason we're narrowing it down to `IOException`? Ideally we'd want to cleanup writer on any exception right?

Reply:
ty